data_store/transformation/pathfinders/pf_transform_r1.py

```diff
@@ -2,14 +2,35 @@
 
 import pandas as pd
 
-from data_store.const import FundTypeIdEnum
-from data_store.transformation.pathfinders.consts import (
-    PF_REPORTING_PERIOD_TO_DATES_HEADERS,
-    PF_REPORTING_ROUND_TO_OBSERVATION_DATES,
-    PF_REPORTING_ROUND_TO_SUBMISSION_DATES,
+from data_store.const import (
+    PF_REPORTING_ROUND_NUMBER_TO_OBSERVATION_DATES,
+    PF_REPORTING_ROUND_NUMBER_TO_SUBMISSION_DATES,
+    FundTypeIdEnum,
 )
+from data_store.transformation.pathfinders.consts import BASE_REPORTING_PERIOD_HEADERS_TO_DATES
 from data_store.transformation.utils import create_dataframe, extract_postcodes
 
+FAS_REPORTING_PERIODS = BASE_REPORTING_PERIOD_HEADERS_TO_DATES.copy()
+FAS_REPORTING_PERIODS.update(
+    {
+        "Financial year 2023 to 2024, (Jan to Mar)": {
+            "start": datetime(2024, 1, 1),
+            "end": datetime(2024, 3, 31, 23, 59, 59),
+        },
+    }
+)
+
+OUTPUT_OUTCOME_REPORTING_PERIOD_HEADERS_TO_DATES = BASE_REPORTING_PERIOD_HEADERS_TO_DATES.copy()
+OUTPUT_OUTCOME_REPORTING_PERIOD_HEADERS_TO_DATES.update(
+    {
+        "Financial year 2023 to 2024, (Jan to Mar)": {
+            "start": datetime(2024, 1, 1),
+            "end": datetime(2024, 3, 31, 23, 59, 59),
+        },
+        "April 2026 and after": {"start": datetime(2026, 4, 1), "end": None},
+    }
+)
+
```

Comment:
I'm not sure I'm convinced that we need to do this yet ... unless you tell me we do because of the upcoming R2 work.

It feels a lot harder to keep track of these dicts when we're referencing dicts defined elsewhere, partially cloning them, and modifying the clones.

But if you say "watch this space" and it'll become apparent with the PFR2 PR, I will trust in Will 🙇

Reply:
Haha yes it is basically because of PF R2, where we basically have this new requirement of having slightly different column header names now for what was "Financial year 2023 to 2024, (Jan to Mar)" in each of the outputs, outcomes and forecast and actual spend tables. As a sneak preview, in `pf_transform_r2.py` we might have this:

```
from XXX import BASE_REPORTING_PERIODS
...

OUTPUT_REPORTING_PERIODS = BASE_REPORTING_PERIODS.copy()
OUTPUT_REPORTING_PERIODS.update(
    {
        "Total cumulative outputs to date, (Up to and including Mar 2024)": {
            "start": None, "end": datetime(2024, 3, 31, 23, 59, 59)
        },
        "April 2026 and after": {
            "start": datetime(2026, 4, 1), "end": None
        }
    }
)

OUTCOME_REPORTING_PERIODS = BASE_REPORTING_PERIODS.copy()
OUTCOME_REPORTING_PERIODS.update(
    {
        "Total cumulative outcomes to date, (Up to and including Mar 2024)": {
            "start": None, "end": datetime(2024, 3, 31, 23, 59, 59)
        },
        "April 2026 and after": {
            "start": datetime(2026, 4, 1), "end": None
        }
    }
)


FAS_REPORTING_PERIODS = BASE_REPORTING_PERIODS.copy()  # Forecast and actual spend
FAS_REPORTING_PERIODS["Total cumulative actuals to date, (Up to and including Mar 2024)"] = {
    "start": None, "end": datetime(2024, 3, 31, 23, 59, 59)
}
```

Definitely open to suggestions here as I'm not hugely proud of the above 🥲

Reply:
One option actually is to rename the columns to match the mapping keys within specific transform functions, and then just maintain a single mapping, including "April 2026 and after" (even though this is not in "Forecast and actual spend"). This would simplify the code

Reply:
Is one option to also embrace the "copy, don't extend" principle we were mildly considering? Should these just be listed in full in a set of consts for each roundm?

Reply:
I'm happy with that tbf. I'll make the change now and if it looks too gross we can always ignore it forever

Reply:
Done